### DISCUSSION TITLE
azd 1.9 changes the container apps environment workload profile

### DISCUSSION THREAD

**Discussion by mip1983** [Azure/azure-dev#3874] [Q&A] [answered]
I've updated to azd 1.9 today and re-run infra synth to deploy my .NET aspire app. It now adds a workload profile to the container apps environment which I only noticed on deploy in my CI/CD env:

```
Deployment Error Details:
EnvironmentNotCreatedWithWorkloadProfile: To add a new workload profile, environment should be created with a Workload Profile
```

It adds these lines to the container app environment config:

```
    workloadProfiles: [{
      workloadProfileType: 'Consumption'
      name: 'consumption'
    }]

```

I've some questions around this change:

1. Is this breaking change intentional?

2. Is there a way of defining in aspire whether you want consumption only or one of the dedicated workload profiles? It changes the cost model and you can't alter after going down a particular path, so it would be good if it didn't make this choice by itself I think.

**Comment by vhvb1989** [comment, marked as answer]
Hi @mip1983 

It was intentional, yes, based on feedback from the Azure Container Apps team. 

There's no way or switch in Aspire to let you choose (hardly opinionated on this for now).  If you want to change it, you need to `infra synth` and manually update it.  You can create a feature request on Aspire repo if you would like to define settings for the managed environment directly from Aspire
